engine/extended_provider/extended_providers.go

```diff
@@ -0,0 +1,164 @@
+package extended_provider
+
+import (
+	"errors"
+
+	"github.com/filecoin-project/storetheindex/api/v0/ingest/schema"
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/libp2p/go-libp2p/core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// ExtendedProvidersAdBuilder contains fields required for building and signing of a new ad with Extended Providers
+type ExtendedProvidersAdBuilder struct {
+	// providerID contains a peer ID of the main provider (the one from the body of the ad)
+	providerID string
+	// privKey contains a private key of the main provider
+	privKey crypto.PrivKey
+	// addrs contains addresses of the main provider
+	addrs []string
+	// providers contains providers' identities, keys, metadata and addresses that will be used as extended providers in the ad.
+	providers []ExtendedProviderInfo
+	// contextID contains optional context id
+	contextID []byte
+	// contextID contains optional metadata
+	metadata []byte
+	// override contains oiverride flag that is false by default
+	override bool
+	// lastAdID contains optional last ad cid which is cid.Undef by default
+	lastAdID cid.Cid
+}
+
+// ExtendedProviderInfo contains information about extended provider.
+type ExtendedProviderInfo struct {
```

Comment:
```suggestion
type Info struct {
```
https://github.com/golang/go/wiki/CodeReviewComments#package-names

Reply:
Changed